backend/src/layer/logic/mealplan_management/meal_plan_manager.rs

```diff
@@ -0,0 +1,103 @@
+use crate::interface::mealplan_management::MensaParseScheduling;
+use crate::interface::mensa_parser::model::ParseCanteen;
+use crate::interface::mensa_parser::MealplanParser;
+use crate::interface::persistent_data::MealplanManagementDataAccess;
+use crate::layer::logic::mealplan_management::relation_resolver::RelationResolver;
+use crate::util::Date;
+use async_trait::async_trait;
+use chrono::Utc;
+use tracing::log::{debug, warn};
+
+pub struct MealPlanManager<Parser, DataAccess>
+where
+    Parser: MealplanParser + Send + Sync,
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+{
+    resolver: RelationResolver<DataAccess>,
+    parser: Parser,
+}
+
+impl<Parser, DataAccess> MealPlanManager<Parser, DataAccess>
+where
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+    Parser: MealplanParser + Send + Sync,
+{
+    pub const fn _new(database: DataAccess, meal_plan_parser: Parser) -> Self {
+        Self {
+            resolver: RelationResolver::_new(database),
```

Comment:
Clippy shows unused constructors even if there are used. So I decided to use _. I'll change that, but then the ci probably fails. Maybe there are ignore macros to prevent this.

Reply:
this is odd... as these are public methods, clippy should not even warn in the first place...